post merge backmerge?

turbomam opened the pull request:
# Guidelines

## _Soft_ Schema Freeze

The `nmdc-schema` and `berkeley-schema-fy24` schemas are under a soft freeze, which means changes **should not** be made that have any downstream implications. To ensure this, all PRs created during the freeze will be closely reviewed with **every** component of the NMDC system in mind.

## Reviewers

To ensure no changes are made unexpectedly, PR creators will use this PR template to tag and notify all task coordinators. Review should be specifically requested from _all_ [Berkeley Schema Roll Out task coordinators](https://docs.google.com/document/d/1XXN1YuaBuSkxPXeiLKm5YxYzXTamBPQrzzeLhlh7PWs/edit#heading=h.u52g8v319adh) that you expect to be affected by this PR.

We expect task coordinators to review PRs and provide feedback/approval within 1 week of when they are identified as reviewers. 

PRs will **NOT** be merged until all task coordinators (or their delegates) have approved it; either here on GitHub (via "`Review changes` > `Approve`" or an equivalent comment) or verbally.

Expedition, questions, and discussion can happen at any meeting.

Delays in review & merging should be addressed in meetings or with NMDC leadership.

| If you expect the changes to<br>impact this component... | ...[request a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review)<br>from this person |
| -- | -- |
| Metadata<br>Schema | @mslarae13 |
| Runtime<br>Mongo database<br>Database migrations | @eecavanna,<br>who will pull in<br>@shreddd as needed |
| Postgres<br>Ingest | @naglepuff |
| Data Portal | @aclum |
| Workflows: MG & MT | @mbthornton-lbl |
| Workflows: MetaB & NOM | @corilo |
| Workflows: LipidO | @ kheal |
| Workflows: MetaP | @SamuelPurvine |
| ETL code | @sujaypatil96 |
| Jupyter notebooks | @brynnz22 |

# PR Information

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation
- [ ] Schema change: Structure and content
  - created, updated, or deleted a `class`, `slot`, or `enum`
  - changed whether a `slot` is `multivalued`
  - changed the way a `slot` is assigned to a `class`
  - changed the `permissible_values` of an `enum`
  - _etc._
- [ ] Schema change: Cleanup and preparation
  - updated the description of a `class`, `slot`, or `enum`
  - updated the `mappings` of a `class`, `slot`, or `enum` to an ontology
  - added an `enum` for _future_ use (it is not in the `range` of any `slot`)
  - _etc._
     
## Description

> _PRs should be [small and concise](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests#write-small-prs)._
>
> Aim to create small, focused pull requests that fulfill a single purpose. Smaller pull requests are easier and faster to review and merge, leave less room to introduce bugs, and provide a clearer history of changes.

- Replace this text with a description of what this PR branch contains. Please keep in mind that all reviewers will be reading this description. Example: "In this branch, I..."

## Related Issues

> All PRs should relate to or fix an issue(s). Please identify the issue(s) below.

- Related Issue(s): #
- Fixes: #

## Did you add/update any tests?

- [ ] Yes
- [ ] No _(Add a justification below)_
- [ ] I need help with writing tests

## Could this schema change make it so any valid data becomes invalid?

> This is a question about what the schema allows. It is not a question about what happens to exists in the NMDC database right now.
> 
> Example: If, in this PR branch, you renamed a `slot` from `foo` to `foo_bar`, the answer to this question would be "yes," **even if** nothing in the NMDC database _currently_ uses the `foo` `slot`.
>
> More examples: `slot` or `class` name changes, changes to a `slot`'s `multivalued` state, changes to a `slot`'s `range` (e.g. string to integer), changes to `slot` assignments to `class`es, changes to an `enum`'s `permissible_values`

- [ ] Yes _(A migrator is required)_
- [ ] No
- [ ] I need help determining this

## If you answered "Yes", does this PR branch include that migrator?

- [ ] Yes
- [ ] No, this PR is incomplete and I need help writing the migrator

## Does this PR have any downstream implications?

> Examples: any change here that requires a change to workflows, workflow automation, the Mongo-to-Postgres ingest process, Jupyter notebooks, the Runtime, etc.

- [ ] Yes _(Explain below)_
- [ ] No

sujaypatil96 committed:
add jinja template for slot documentation pages

sujaypatil96 committed:
typo in slot documentation page note

sujaypatil96 committed:
change abstract slot documentation page banner to 'populated'

kheal committed:
Change metagenom_annotation_id slot to was_generated_by on FunctionalAnnotationAggMember

kheal committed:
Make was_generated_by requiured on FunctionalAnnotationAggMember through slot usage

kheal committed:
Add migrator for FunctionalAnnotationAggMember slot move

kheal committed:
Move count requirement from slot to rule

kheal committed:
Fix rule on FunctionalAnnotationAggMember

kheal committed:
Add invalid example to test rule on FunctionalAnnotationAggMember

kheal committed:
Remove count rule from FunctionalAnnotationAggMember, leaving requirement

kheal committed:
Update tests

kheal committed:
Merge branch 'main' into 1253_metap_aggtable

kheal committed:
Update metatranscriptome functional aggregation example

kheal committed:
Fix count slot requirement

kheal committed:
Fix typo in nmdc.yaml

kheal committed:
Merge main into 1253 branch

kheal committed:
Add unique_keys to FunctionalAnnotationAggMember

kheal committed:
Add unique_keys to FunctionalAnnotationAggMember

turbomam committed:
v10.9.1

turbomam committed:
Merge branch 'main' into resolve-merge-conflict-in-nmdc.py

turbomam committed:
Merge pull request #2195 from microbiomedata/resolve-merge-conflict-in-nmdc.py

Merge Berkeley schema (`berkeley-schema-fy24/main`) into Legacy schema (`nmdc-schema/main`)

turbomam committed:
v11.0.0

pkalita-lbl committed:
Add `bulk chemistry` back to `AnalysisTypeEnum`

pkalita-lbl committed:
Merge pull request #2197 from microbiomedata/add-bulk-chemistry-pv

Add `bulk chemistry` back to `AnalysisTypeEnum`

pkalita-lbl committed:
11.0.1

turbomam committed:
moved problem example tp problem dir

kheal committed:
Merge main into 1253_metap_aggtable

kheal committed:
Add description for count slot on FunctionalAnnotationAggMember

kheal committed:
Remove count description on class and rely on slot definition

kheal committed:
Clean up functional aggregation migrator

bmeluch committed:
invalid dissolvingprocess now tests invalid substances_used

turbomam committed:
pyproject.toml spelling fixes

turbomam committed:
but no formatting changes

bmeluch committed:
Merge pull request #2201 from microbiomedata/2200-srcdatainvaliddissolvingprocess-minimal-no-solventyaml-implies-solvent-is-required-testing-fails

move valid `src/data/invalid/DissolvingProcess-minimal-no-solvent.yaml` -> `src/data/problem/invalid/DissolvingProcess-minimal-no-solvent.yaml`

aclum committed:
Add figshare as a doi provider

aclum committed:
Update Database-study-set-with-dois.yaml

Add example with figshare as doi_provider

aclum committed:
Add kegg module prefix

sujaypatil96 committed:
Merge branch 'main' into custom-slot-jinja

turbomam committed:
don't use https://api-berkeley.microbiomedata.org in makefiles any more

turbomam committed:
don't use https://api-berkeley.microbiomedata.org in sparql any more

turbomam committed:
Merge pull request #2213 from microbiomedata/2212-replace-usages-of-api-berkeley-in-makefiles-with-production-api

2212 replace usages of api berkeley in makefiles with production api

turbomam committed:
no warning from poetry run date-created-blank-node --help

turbomam committed:
no more scalar in_subsets

turbomam committed:
Merge pull request #2218 from microbiomedata/1464-in_subset-slot-should-be-used-as-multivalued-throughout

no more scalar in_subsets

turbomam committed:
Merge pull request #2209 from microbiomedata/2196-comment-in-pyprojecttoml-contains-a-typo-turtle

Fix spelling errors in `pyproject.toml`

sujaypatil96 committed:
Merge pull request #2140 from microbiomedata/custom-slot-jinja

Add custom jinja template for slot documentation pages

turbomam committed:
why are some collectiosn omitted or modified?

turbomam committed:
Merge pull request #2220 from microbiomedata/2219-update-documentation-of-workaroundscollection-exclusion-in-pure-export-in-projectmakefile

why are some collectiosn omitted or modified?

aclum committed:
Merge pull request #2210 from microbiomedata/2192-add-figshare-to-doiproviderenum

Add figshare as a permissible value to doi provider enumeration

aclum committed:
Update nmdc.yaml

Uppercasing prefix for KEGG.MODULE at the recommendation of Chris.

aclum committed:
updating example data for MagsAnalysis

aclum committed:
make members_id multivalued

aclum committed:
remove swp file

turbomam committed:
regeneration and really invalid dissolving file

turbomam committed:
merge latest release into main for doc gen

aclum committed:
Merge pull request #2211 from microbiomedata/2169-add-kegg-module-prefix

Add kegg module prefix

aclum committed:
Add permissible values to ProcessingInstitutionEnum

Add Azenta as a permissible value

aclum committed:
Create NucleotideSequencing-valid.yaml

aclum committed:
updates to MagsAnalysis slots

Add descriptions and mappings, add enum for bin quality range.

aclum committed:
fix typo

aclum committed:
Update core.yaml

poetry doesn't like my todo statements

aclum committed:
Update core.yaml

todos need to be a list

aclum committed:
Update core.yaml

todo should be todos

aclum committed:
Update core.yaml

Adding minimum values for MagsAnalysis slots

aclum committed:
Update core.yaml

aclum committed:
Create MagsAnalysis-exhaustive.yaml

aclum committed:
Create MagsAnalysis-invalid-negative-int.yaml

aclum committed:
Create MagsAnalysis-invalid-float-for-16s.yaml

aclum committed:
Create MagsAnalysis-invalid-missing-bin-name.yaml

aclum committed:
Update MagsAnalysis-invalid-negative-int.yaml

add type

aclum committed:
Update MagsAnalysis-invalid-float-for-16s.yaml

aclum committed:
Update MagsAnalysisActivity-invalid_ncbi_lineage_tax_ids.yaml

aclum committed:
Create MagsAnalysis-invalid-bin_qulity.yaml

aclum committed:
Update MagsAnalysis-exhaustive.yaml

aclum committed:
Update core.yaml

aclum committed:
Update workflow_execution_activity.yaml

Adding slot description and minimum values.

aclum committed:
Update workflow_execution_activity.yaml

mslarae13 commented:
@turbomam you have the soft freeze PR template above, but didn't edit any of the text or fill it out. 
I missed some stuff last week, so what is this for?

aclum committed:
Update MagsAnalysis-exhaustive.yaml

aclum committed:
Update MagsAnalysis-invalid-bin_qulity.yaml

aclum committed:
Update MagsAnalysis-invalid-float-for-16s.yaml

aclum committed:
Update MagsAnalysis-invalid-missing-bin-name.yaml

aclum committed:
Update MagsAnalysis-invalid-negative-int.yaml

aclum committed:
Update MagsAnalysisActivity-invalid_ncbi_lineage_tax_ids.yaml

aclum committed:
make sure example invalid MagsAnalysis fail for only one reason

aclum committed:
Mags valid example data

turbomam committed:
won't leave in pre_vs_berkeley/

turbomam committed:
prep for mkdocs

turbomam committed:
Merge pull request #2224 from microbiomedata/634-Azenta

634 azenta

turbomam committed:
v10 to v11 now in left with images

turbomam committed:
reason for major update in fork

aclum committed:
Update core.yaml

Moving comments to see_also with doi for gtdb-tk slots

aclum committed:
Add genomad files to FileTypeEnum

aclum committed:
Merge pull request #2228 from microbiomedata/aclum-patch-2

Add genomad files to FileTypeEnum

aclum committed:
Create DataObject-genomad.yaml

aclum committed:
change valid file to Database instead of DataObject

aclum committed:
remove DataObject-genomad.yaml

kheal committed:
Add scaffold of manifest modeling

kheal committed:
Add example for using Manifest class

kheal committed:
Add more permissible values for CollectionTypeEnum

kheal committed:
Change collection_type to manifest_category

kheal committed:
Change collection_type to manifest_category

kheal committed:
Move in_manifest to DataObject from DataGeneration

kheal committed:
Fix mass spec gc example data

kheal committed:
Edit description for Manifest and ManifestCategoryEnum values

kheal committed:
Fix typo

aclum committed:
Merge pull request #2229 from microbiomedata/2182-add-genomad-file-type-enumerations

2182 add genomad file type enumerations

aclum committed:
Merge pull request #2225 from microbiomedata/2160-other-updates-to-magbin-modeling

updates to MagsAnalysis slots

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1819802989

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1819964644

kheal committed:
Update name of migrator for functional_annotation_agg slot name change

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1820944288

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1820953055

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1820956940

Commit:
move has_calibration off of WorkflowExecution subclasses

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1820940055

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1820940055 part 2

turbomam committed:
https://github.com/microbiomedata/nmdc-schema/pull/2227#discussion_r1820948629

turbomam committed:
Update src/docs/v10-vs-v11-retrospective.md

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

turbomam committed:
selected legacy parents explanation

turbomam committed:
Merge branch '2226-move-markdown-and-images-from-httpsgithubcommicrobiomedataberkeley-schema-fy24-into-this-repos-docs-directory' of github.com:microbiomedata/nmdc-schema into 2226-move-markdown-and-images-from-httpsgithubcommicrobiomedataberkeley-schema-fy24-into-this-repos-docs-directory

turbomam committed:
use "database" after MongoDB when necessary

turbomam committed:
describe updates to Extraction slots

turbomam committed:
specifically say that there were changed in Extraction *slots*

turbomam committed:
Merge pull request #2227 from microbiomedata/2226-move-markdown-and-images-from-httpsgithubcommicrobiomedataberkeley-schema-fy24-into-this-repos-docs-directory

v10 vs v11/berkeley documentation

Commit:
add first pass at migrator

Commit:
update doc strings

Commit:
fix string literal

Commit:
rename migrator

Commit:
change has_calibration range

Commit:
fix doc test

Commit:
update doc tests

Commit:
fix migrator and implement partial

Commit:
add no op migrator

Commit:
implement partials

Commit:
fix documentation

Commit:
remove extraneous code

Commit:
remove notes

mslarae13 committed:
delete PR template

Commit:
Update nmdc_schema/migrators/partials/migrator_from_11_0_3_to_11_1_0/__init__.py

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

Commit:
Update nmdc_schema/migrators/partials/migrator_from_11_0_3_to_11_1_0/migrator_from_11_0_3_to_11_1_0_part_1.py

Thanks!!

Co-authored-by: eecavanna <134325062+eecavanna@users.noreply.github.com>

Commit:
change variable name

Commit:
reformat and change to name

eecavanna committed:
Replace version number and generate derivatives before building docs

eecavanna committed:
Generate inter-collection relationship diagram and add it to docs site

eecavanna committed:
Regenerate `poetry.lock` file by running `$ poetry lock --no-update`

eecavanna committed:
Make comment more accurate

eecavanna committed:
Use `poetry` instead of `pipx` (`pipx` is not otherwise used)

turbomam committed:
Merge pull request #2241 from microbiomedata/2240-add-inter-collection-diagram-generated-by-refscan-to-schema-documentation

Include inter-collection relationship graph in schema documentation

turbomam committed:
Merge pull request #2235 from microbiomedata/calibration-change

Move `has_calibration` from `workflow_execution_set` to `data_generation_set`

turbomam committed:
Merge pull request #2237 from microbiomedata/2236-remove-pr-template

Delete PR template

kheal committed:
Rename migrator for merge prep

kheal committed:
Merge main into 1253_metap_aggtable

kheal committed:
Convert functional annotation agg to partial migrator

kheal committed:
Modify partial migrator init

samobermiller committed:
removed direct_infusion_category and DirectInfusionEnum from  v10-vsv11-retrospective.md and basic_slots.yaml

turbomam committed:
Merge pull request #2248 from microbiomedata/Remove-direct_infusion_category-and-directinfusionenum-#2234

Remove `direct_infusion_category` and `DirectInfusionEnum`

kheal committed:
Move count description to slot_usage

kheal committed:
Add test for unique_keys for distribution

kheal committed:
Add description and comment to in_manifest slot

kheal committed:
Update descriptions for Manifest class and associated Enum

kheal committed:
Fix invalid functional_annotation_agg example

kheal committed:
Replace technical replicates with poolable replicates

kheal committed:
Merge pull request #2230 from microbiomedata/manifest

Add `Manifest` class and associated slots and Enum

kheal committed:
Deprecate ProteinQuantification and PeptideQuantification and slots used only in these classes

kheal committed:
Add migrator for removing has_peptide_quantificiations slot

kheal committed:
Improve migrator for has_peptide_quantifications

kheal committed:
Fix typo in migrator

kheal committed:
Convert migrator to partial

kheal committed:
Remove commands for removing has_peptide_quantifications from records
Fixes #2231

kheal committed:
Add versions to migrators

eecavanna commented:
Hi @turbomam, are you OK with me closing this PR? I get a notification whenever commits get added to `nmdc-schema/main` because that branch is involved in this PR and my GitHub username is tagged in the PR description.

kheal committed:
Remove unique_keys from FunctionalAnnotationAggMember class

kheal committed:
Merge pull request #2203 from microbiomedata/1253_metap_aggtable

Update `FunctionalAnnotationAggMember` class for compatibility with MetaP Aggregation tables and implement migrator

kheal committed:
Rename migrator to be part 3

kheal committed:
Resolve merge conflicts

eecavanna committed:
Update `_from_version` and `_to_version` to reflect role of migrator

eecavanna committed:
Update `_from_version` to indicate preceding partial migrator

eecavanna committed:
Merge pull request #2254 from microbiomedata/2253-migration-update-migrator-_to_version-to-end-in-part_2

kheal committed:
Add test for deprecated class

eecavanna committed:
Delete deprecation-specific example data file

This effectively reverts commit d5da1663e0ee105d4dd39f6079f105b09dfedc58

eecavanna committed:
Define abstract adapter method that copies a value between two fields

eecavanna committed:
Implement dictionary adapter method that copies value between two fields

eecavanna committed:
Merge pull request #2251 from microbiomedata/2250_remove_protein_results

Remove `has_peptide_quantifications`, deprecate associated classes and slots, and implement migrator

eecavanna committed:
Implement Mongo adapter method and a test that targets it

eecavanna committed:
Reorder doctests in dictionary adapter method to be more intuitive

eecavanna committed:
Implement adapter method that deletes field from all documents in coll.

eecavanna committed:
Merge branch '2261-migrations-implement-adapter-method-delete_field_from_each_document' into 2264-migrations-update-migrators-to-use-newly-implemented-adapter-methods-for-copying-and-deleting

eecavanna committed:
Update `part_2` migrator to delegate iteration to MongoDB (was Python)

eecavanna committed:
Merge pull request #2262 from microbiomedata/2259-migrations-implement-adapter-method-copy_value_from_field_to_field_in_each_document

Migrations: Implement adapter method that copies values between two fields in all documents in a collection

eecavanna committed:
Merge pull request #2263 from microbiomedata/2261-migrations-implement-adapter-method-delete_field_from_each_document

Migrations: Implement adapter method that deletes field from all documents in collection

eecavanna committed:
Merge pull request #2265 from microbiomedata/2264-migrations-update-migrators-to-use-newly-implemented-adapter-methods-for-copying-and-deleting

Migrations: Update migrator to run faster by delegating document iteration to MongoDB

eecavanna committed:
Generate release artifacts by running `$ make squeaky-clean all test`

eecavanna committed:
Fix HTML validation issues in `README.md`

eecavanna committed:
11.1.0

Merge pull request #2266 from microbiomedata/2252-release-nmdc-schema-1110

Generate release artifacts for NMDC Schema version `11.1.0`

aclum commented:
Can this draft PR be closed? It is on berkeley-schema-fy24. @turbomam @eecavanna @kheal

eecavanna commented:
I'm in favor of closing it.

kheal commented:
I have no stake in this PR.

sierra-moxon committed:
re-add prefixmaps generation to makefile

sierra-moxon committed:
re-add prefixmaps generation to makefile

sierra-moxon committed:
Merge pull request #2268 from microbiomedata/prefixmap

re-add prefixmaps generation to makefile

aclum commented:
closing this PR cc @turbomam
